### DISCUSSION TITLE
Excluding from TypeScript

### DISCUSSION THREAD

**Discussion by malyutki** [spatie/typescript-transformer#37] [Q&A]
Hi, is it any way to exclude some properties from transforming to TypeScript?
I want to add some variables from route, but I don't want to show them in newly created TypeScript

**Comment by rubenvanassche** [comment]
Not at the moment, but you can create a new transformer based upon the DTO transformer which implements this feature.

**Comment by sneakylenny** [comment]
For anyone wondering, you can do this without writing a transformer by doing this:



```php
// ...
use Spatie\LaravelData\Attributes\WithoutValidation; // Skips validation when creating (optional)
use Spatie\TypeScriptTransformer\Attributes\Hidden; // Excludes it from generated TypeScript
// ...

#[TypeScript()]
class CreatePostData extends Data
{
    #[Hidden, WithoutValidation] // Apply here
    public string $hideMe;
}
```

`$hideMe` will now be excluded from the generated TypeScript
